packages/services/src/competition.service.ts

```diff
@@ -1306,6 +1372,286 @@ export class CompetitionService {
+          `Games must have end times and winners before the competition can end. Details: ${details}`,
+      );
+    }
+
+    this.logger.debug(
+      `[CompetitionService] Validated ${finalGames.length} final games are ready for scoring`,
+    );
+  }
+
+  /**
+   * Ensure all final games in an NFL competition are scored
+   * This validates all games are finalized and scored, else, throws an error.
+   * @param competitionId The competition ID
+   * @returns void
+   * @throws Error if any games are not finalized or scored
+   */
+  async #ensureNflGamesScored(competitionId: string): Promise<void> {
+    // Get all games for this competition
+    const gameIds =
+      await this.sportsService.competitionGamesRepository.findGameIdsByCompetitionId(
+        competitionId,
+      );
+
+    // Get game details to check which are final
+    const games = await this.sportsService.gamesRepository.findByIds(gameIds);
+    const finalGames = games.filter((game) => game.status === "final");
+
+    this.logger.debug(
+      `[CompetitionService] Ensuring ${finalGames.length} final games are scored for competition ${competitionId}`,
+    );
+
+    let scoredCount = 0;
+    let alreadyScoredCount = 0;
+    const errors: Array<{ gameId: string; error: unknown }> = [];
+
+    // Score each final game (note: `scoreGame` is idempotent)
+    for (const game of finalGames) {
+      try {
+        const agentsScoredCount =
+          await this.sportsService.gameScoringService.scoreGame(game.id);
+
+        if (agentsScoredCount > 0) {
+          scoredCount++;
+          this.logger.debug(
+            `[CompetitionService] Scored game ${game.id} (${game.awayTeam} @ ${game.homeTeam}): ${agentsScoredCount} agents`,
+          );
+        } else {
+          alreadyScoredCount++;
+        }
+      } catch (error) {
+        errors.push({ gameId: game.id, error });
+        this.logger.error(
+          { error, gameId: game.id },
+          `[CompetitionService] Error scoring game ${game.id} (${game.awayTeam} @ ${game.homeTeam})`,
+        );
+      }
+    }
+
+    // If any games failed to score after validation, this is a serious error
+    if (errors.length > 0) {
+      throw new Error(
+        `Failed to score ${errors.length} game(s) for competition ${competitionId}. ` +
+          `This should not happen after validation. Game IDs: ${errors.map((e) => e.gameId).join(", ")}`,
+      );
+    }
+
+    this.logger.info(
+      `[CompetitionService] Game scoring complete for competition ${competitionId}: ` +
+        `${scoredCount} newly scored, ${alreadyScoredCount} already scored`,
+    );
+  }
+
+  /**
+   * End a competition for NFL competitions only
+   * @param competitionId The competition ID
+   * @returns The updated competition and final NFL-specific leaderboard
+   */
+  async endNflCompetition(competitionId: string): Promise<{
```

Comment:
this is a bit duplicative with `endCompetition`, but it's needed because the logic and response object differs.